src/io/functions/iologindata_load_player.cpp

```diff
@@ -9,9 +9,774 @@
 
 #include "pch.hpp"
 
+#include "creatures/players/wheel/player_wheel.hpp"
 #include "io/functions/iologindata_load_player.hpp"
+#include "game/game.h"
+
+bool IOLoginDataLoad::preLoadPlayer(Player* player, const std::string &name) {
+	Database &db = Database::getInstance();
+
+	std::ostringstream query;
+	query << "SELECT `id`, `account_id`, `group_id`, `deletion`, (SELECT `type` FROM `accounts` WHERE `accounts`.`id` = `account_id`) AS `account_type`";
+	query << ", (SELECT `premdays` FROM `accounts` WHERE `accounts`.`id` = `account_id`) AS `premium_days`";
+	query << ", (SELECT `premdays_purchased` FROM `accounts` WHERE `accounts`.`id` = `account_id`) AS `premium_days_purchased`";
+	query << ", (SELECT `creation` FROM `accounts` WHERE `accounts`.`id` = `account_id`) AS `creation_timestamp`";
+	query << " FROM `players` WHERE `name` = " << db.escapeString(name);
+	DBResult_ptr result = db.storeQuery(query.str());
+	if (!result) {
+		return false;
+	}
+
+	if (result->getNumber<uint64_t>("deletion") != 0) {
+		return false;
+	}
+
+	player->setGUID(result->getNumber<uint32_t>("id"));
+	Group* group = g_game().groups.getGroup(result->getNumber<uint16_t>("group_id"));
+	if (!group) {
+		SPDLOG_ERROR("Player {} has group id {} which doesn't exist", player->name, result->getNumber<uint16_t>("group_id"));
+		return false;
+	}
+	player->setGroup(group);
+	player->accountNumber = result->getNumber<uint32_t>("account_id");
+	player->accountType = static_cast<account::AccountType>(result->getNumber<uint16_t>("account_type"));
+	if (!g_configManager().getBoolean(FREE_PREMIUM)) {
+		player->premiumDays = result->getNumber<uint16_t>("premium_days");
+	} else {
+		player->premiumDays = std::numeric_limits<uint16_t>::max();
+	}
```

Comment:
Loyalty/VIP systems rely on this number, FREE_PREMIUM is just a trigger to send to the client and to enable spells/boat etc.
```suggestion
	player->premiumDays = result->getNumber<uint16_t>("premium_days");
```